### PR TITLE
feat(website): Added starlight docs for website, rebuilt Grid using flexbox

### PR DESCRIPTION
**Pull request by c4h9ris**

### CODE REVIEW

**Comment by bcheidemann** [comment]
Looks like [astro-live-code](https://github.com/mattjennings/astro-live-code) doesn't have a license. I think this is probably fine though since we're only using it for the documentation site.

File: website/src/content/docs/index.mdx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+title: Welcome to Gemini UI
+description: An open source component library for Astro. Build beautiful stellar user interfaces with minimal effort.
+template: splash
+hero:
+  tagline: An open source component library for Astro. Build beautiful stellar user interfaces with minimal effort.
+  image:
+    file: ../../assets/gemini-logo.png
+  actions:
+    - text: Get Started
+      link: /guides/example/
```
Comment (bcheidemann) [inline website/src/content/docs/index.mdx:11]:
```suggestion
      link: /guides/getting-started/
```

File: website/src/content/docs/index.mdx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      icon: right-arrow
+      variant: primary
+    - text: View on Github
+      link: https://github.com/gemini-ui/gemini-ui-monorepo
+      icon: external
```
Comment (bcheidemann) [inline website/src/content/docs/index.mdx:16]:
```suggestion
    - text: View on Github
      link: https://github.com/gemini-ui/gemini-ui-monorepo
      icon: external
      attrs:
        target: _blank
```

File: website/src/content/docs/guides/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+description: A guide to setting up Gemini UI
+---
+
+import { Tabs, TabItem } from "@astrojs/starlight/components";
+
+To install Gemini UI run the following command
+
+<Tabs>
+  <TabItem label="npm">``` npm i @gemini-ui-astro/components ```</TabItem>
+  <TabItem label="Yarn">``` yarn add @gemini-ui-astro/components ```</TabItem>
```
Comment (bcheidemann) [inline website/src/content/docs/guides/getting-started.mdx:12]:
```suggestion
  <TabItem label="npm">```sh npm i @gemini-ui-astro/components ```</TabItem>
  <TabItem label="Yarn">```sh yarn add @gemini-ui-astro/components ```</TabItem>
```

Does this work?

File: website/src/layouts/Layout.astro [hunk 4]
```diff
@@ -1,56 +0,0 @@
...
-<!doctype html>
-<html lang="en">
-  <head>
-    <meta charset="UTF-8" />
-    <meta name="description" content="Astro description" />
-    <meta name="viewport" content="width=device-width" />
-    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
-    <meta name="generator" content={Astro.generator} />
-    <title>{title}</title>
-    <DynamicGlow />
```
Comment (bcheidemann) [inline website/src/layouts/Layout.astro:21]: I don't see where this is being imported in the docs website but it seems to be working anyway somehow 🤔 any idea how this is working?

File: website/src/content/docs/components/image.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+<Image
+  class="neom-1"
+  src="/neom-400x300.webp"
+  alt="Sandstone plateau, Hisma Desert - NEOM, Saudi Arabia"
+  width="400"
+  caption
+  dynamicGlow
+>
+  Figure 2: The vibrant waters of NEOM, are home to some of the most picturesque
+  and pristine islands in the Red Sea | Islands of NEOM - NEOM, Saudi Arabia.
```
Comment (bcheidemann) [inline website/src/content/docs/components/image.mdx:42]: Not sure if this was already the case but the caption doesn't match the image here

File: website/src/content/docs/components/card.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+## Usage
+
+```astro live
+---
+import { Card } from "@gemini-ui-astro/components";
+import { Button } from "@gemini-ui-astro/components";
+---
+
+<Card class="example-card" dynamicGlow style={{ marginBottom: "32px" }}>
```
Comment (bcheidemann) [inline website/src/content/docs/components/card.mdx:16]: We can fix this later but the card text has low contrast with the background in light mode.

**Review by bcheidemann — APPROVED** [review]
🔥

This is great! Huge upgrade =)

File: website/src/content/docs/index.mdx (hunk 1 above)
Comment (c4h9ris) [reply website/src/content/docs/index.mdx:11]: Good catch

File: website/src/content/docs/guides/getting-started.mdx (hunk 3 above)
Comment (c4h9ris) [reply website/src/content/docs/guides/getting-started.mdx:12]: What were you hoping it would do ?

File: website/src/layouts/Layout.astro (hunk 4 above)
Comment (c4h9ris) [reply website/src/layouts/Layout.astro:21]: I have no idea actually

File: website/src/content/docs/components/card.mdx (hunk 6 above)
Comment (c4h9ris) [reply website/src/content/docs/components/card.mdx:16]: Yeah we need to implement a light / dark mode

File: website/src/content/docs/guides/getting-started.mdx (hunk 3 above)
Comment (bcheidemann) [reply website/src/content/docs/guides/getting-started.mdx:13]: Add a copy button 😁

**Pull request merged**